src/Model/ContentNegotiationEntity.php

```diff
@@ -10,13 +10,10 @@
 class ContentNegotiationEntity
 {
     /**
-     * @param string $name
      * @param array<string, mixed> $config
      */
-    public function __construct($name, $config)
+    public function __construct(public string $name, public array $config)
```

Comment:
I see a number of these changes in the various entity classes (adding constructor property promotion with explicit types).

Overall, I think they are fine; these were the documented types in annotations, and what we test against; further, using constructor property promotion ensures the properties are actually declared, which they were not before.

However, it _is_ a subtle BC break for anybody extending these classes. @Ocramius , what are your thoughts? Should we go ahead with these for next minor, push to next major, or ask to revert them?

Reply:
On the constructor, they are mostly fine due to them not affecting LSP checks

Reply:
since the property is dynamic, it is bc break https://3v4l.org/vLe8v

Rector can cover that:

1. dynamic: skipped https://getrector.com/demo/9cbff805-62d4-4557-baab-ea1875b11127
2. defined property: applied 
      a. typed https://getrector.com/demo/1ef76619-2b7f-43d4-9c7b-0cd009d69b45
      b. not typed https://getrector.com/demo/085e9f92-ed93-45b4-ace9-1673c8770816

Instead, I suggest add `public` property instead:

```php
public $name;
public $config;
```

Reply:
anyway, Rector has `CompleteDynamicPropertiesRector` to auto add public property if needed:

https://getrector.com/demo/a8f2c237-a12c-4fae-9f92-eba6677a53d4

Reply:
So, if you combine ` CompleteDynamicPropertiesRector` + `ClassPropertyAssignToConstructorPromotionRector `, you will get it processed but kept the `@param` without adding new type

https://getrector.com/demo/26c0e45b-a208-4a5f-acd8-f82e11bbda51

Reply:
On summary, just remove the type, and it will be safer:

```suggestion
    public function __construct(public $name, public $config)
```

Reply:
> since the property is dynamic, it is bc break

Right, but these are (a) essentially internal, and (b) the expectation is that the properties are present. I'd argue it actually tightens the contract and makes it more reliable.

Reply:
@weierophinney I see, it need `@internal` flag then

Reply:
> remove the type,

We should have been validating the type all along, however. This ensures the type is correct.